### PR TITLE
Report subprocess death from signal to the user

### PR DESCRIPTION
**Pull request by sbc100**
Fixes: #14275

### CODE REVIEW

File: tools/shared.py [hunk 1]
```diff
@@ -117,6 +118,15 @@ def mp_run_process(command_tuple):
   return ret
 
 
+def returncode_to_str(code):
+  assert code != 0
+  if code < 0:
+    signal_name = signal.Signals(-code).name
+    return f'received {signal_name}'
```
Comment (kripken) [inline tools/shared.py:125]:
```suggestion
    return f'received {signal_name} ({code})'
```

File: tests/test_other.py [hunk 2]
```diff
@@ -10542,3 +10542,15 @@ def test_special_chars_in_arguments(self):
...
+  def test_report_subprocess_signals(self):
+    # Test that when subprocess is killed by signal we report the signal name
+    create_file('die.sh', '''\
+#!/bin/sh
+kill -9 $$
+    ''')
+    make_executable('die.sh')
+    with env_modify({'EM_COMPILER_WRAPPER': './die.sh'}):
+      err = self.expect_fail([EMCC, test_file('hello_world.c')])
+      self.assertContained('failed (received SIGKILL)', err)
```
Comment (kripken) [inline tests/test_other.py:10556]:
```suggestion
      self.assertContained('failed (received SIGKILL (-9))', err)
```

File: tools/shared.py [hunk 3]
```diff
@@ -188,7 +198,8 @@ def get_finished_process():
             logger.info(out)
           if err:
             logger.error(err)
-          raise Exception('Subprocess %d/%d failed with return code %d! (cmdline: %s)' % (idx + 1, len(commands), finished_process.returncode, shlex_join(commands[idx])))
+
+          raise Exception('Subprocess %d/%d failed (%s)! (cmdline: %s)' % (idx + 1, len(commands), returncode_to_str(finished_process.returncode), shlex_join(commands[idx])))
```
Comment (kripken) [inline tools/shared.py:202]: This isn't tested, and I'm not sure how to test it in a reasonable way. But I'd recommend at least testing it manually locally before landing (say adding a throw in the system libs code, and building a system lib). Just to rule out a typo on this line that ends up masking the proper reported error.

**Review by kripken — APPROVED** [review]

File: tools/shared.py (hunk 3 above)
Comment (sbc100) [reply tools/shared.py:202]: Done.

**Pull request merged**